code/game/machinery/computer/slotmachine.dm

```diff
@@ -2,74 +2,82 @@
 	density = TRUE
+	clicksound = null
 	use_power = 1
 	idle_power_usage = 250
 	active_power_usage = 500
 	circuit = /obj/item/circuitboard/slot_machine
 	var/emmaged = FALSE
+	light_color = LIGHT_COLOR_BROWN
 	var/money = 3000 //How much money it has CONSUMED
 	var/plays = 0
-	var/working = 0
+	var/working = FALSE
 	var/balance = 0 //How much money is in the machine, ready to be CONSUMED.
 	var/jackpots = 0
+	var/paymode = CREDITCHIP //toggles between CREDITCHIP/COIN, defined above
+	var/cointype = /obj/item/coin/iron //default cointype
 	var/list/coinvalues = list()
 	var/list/reels = list(list("", "", "") = 0, list("", "", "") = 0, list("", "", "") = 0, list("", "", "") = 0, list("", "", "") = 0)
-	var/list/symbols = list(SEVEN = 1, "<font color='orange'>&</font>" = 2, "<font color='yellow'>@</font>" = 2, "<font color='green'>$</font>" = 2, "<span class='notice'>?</span>" = 2, "<font color='grey'>#</font>" = 2, "<font color='white'>!</font>" = 2, "<font color='fuchsia'>%</font>" = 2) //if people are winning too much, multiply every number in this list by 2 and see if they are still winning too much.
+	var/list/symbols = list(SEVEN = 1, "<font color='orange'>&</font>" = 2, "<font color='yellow'>@</font>" = 2, "<font color='green'>$</font>" = 2, "<font color='blue'>?</font>" = 2, "<font color='grey'>#</font>" = 2, "<font color='white'>!</font>" = 2, "<font color='fuchsia'>%</font>" = 2) //if people are winning too much, multiply every number in this list by 2 and see if they are still winning too much.
 
-	light_color = LIGHT_COLOR_BROWN
 
 /obj/machinery/computer/slot_machine/Initialize()
 	. = ..()
 	jackpots = rand(1, 4) //false hope
 	plays = rand(75, 200)
+	money = round(rand(2500, 3500), 5)
 
-	toggle_reel_spin(1) //The reels won't spin unless we activate them
+	toggle_reel_spin(TRUE) //The reels won't spin unless we activate them
 
 	var/list/reel = reels[1]
-	for(var/i = 0, i < reel.len, i++) //Populate the reels.
+	for(var/i in 1 to reel.len) //Populate the reels.
 		randomize_reels()
 
-	toggle_reel_spin(0)
+	toggle_reel_spin(FALSE)
 
-	for(var/cointype in typesof(/obj/item/coin))
-		var/obj/item/coin/C = cointype
-		var/value = get_value(C)
-		coinvalues["[cointype]"] = value
+	for(cointype in typesof(/obj/item/coin))
+		var/obj/item/coin/C = new cointype
+		coinvalues["[cointype]"] = get_value(C)
+		qdel(C) //Sigh
 
 /obj/machinery/computer/slot_machine/Destroy()
 	if(balance)
-		give_coins(balance)
+		give_payout(balance)
 	return ..()
 
-/obj/machinery/computer/slot_machine/process()
-	. = ..()
+/obj/machinery/computer/slot_machine/process(delta_time)
+	. = ..() //Sanity checks.
 	if(!.)
 		return .
 
-	money++
+	money += round(delta_time / 2) //SPESSH MAJICKS
 
 /obj/machinery/computer/slot_machine/update_icon()
-	if(stat & NOPOWER)
-		icon_state = "slots0"
-
-	else if(stat & BROKEN)
-		icon_state = "slotsb"
-
-	else if(working)
-		icon_state = "slots2"
-
+	if(stat & BROKEN)
+		icon_state = "slots_broken"
+		if(stat & NOPOWER)
```

Comment:
This condition (`stat & NOPOWER`) unless the previous condition is also triggered (`stat & BROKEN`)

Shouldn't be first condition be something like `(stat & BROKEN || stat & NOPOWER)` ?